mplc/datasets/dataset_titanic.py

```diff
@@ -141,3 +143,74 @@ def train_val_split_global(x, y):
 
 def train_test_split_global(x, y):
     return train_test_split(x, y, test_size=0.1, random_state=42)
+
+
+class LogisticRegression(skLR):
+    def __init__(self):
+        super(LogisticRegression, self).__init__(max_iter=10000, warm_start=1, random_state=0)
+        self.coef_ = None
+        self.intercept_ = None
+
+    def fit(self, x_train,
+            y_train,
+            batch_size,
+            epochs,
+            verbose,
+            validation_data):
+        history = super(LogisticRegression, self).fit(x_train, y_train)
+        [loss, acc] = self.evaluate(x_train, y_train)
+        [val_loss, val_acc] = self.evaluate(*validation_data)
+        # Mimic Keras' history
+        history.history = {
+            'loss': [loss],
+            'accuracy': [acc],
+            'val_loss': [val_loss],
+            'val_accuracy': [val_acc]
+        }
+
+        return history
+
+    def evaluate(self, x_eval, y_eval, **kwargs):
+        if self.coef_ is None:
+            model_evaluation = [0] * 2
+        else:
+            loss = log_loss(y_eval, self.predict(x_eval))  # mimic keras model evaluation
+            accuracy = self.score(x_eval, y_eval)
+            model_evaluation = [loss, accuracy]
+
+        return model_evaluation
+
+    def save_weight(self, path):
+        if self.coef_ is None:
+            raise ValueError('The model has never been fit, coef and intercept are set to None')
```

Comment:
I would soften the message, like: "Coef and intercept are set to None, it seems the model has not been fit properly."
```suggestion
    def save_weight(self, path):
        if self.coef_ is None:
            raise ValueError('Coef and intercept are set to None, it seems the model has not been fit properly.')
```